modules/virt-configuring-vm-dpdk.adoc

```diff
@@ -0,0 +1,109 @@
+
+:_content-type: PROCEDURE
+[id="virt-configuring-vm-dpdk_{context}"]
+= Configuring a virtual machine for DPDK workloads
+
+You can run Data Packet Development Kit (DPDK) workloads on virtual machines (VMs) to achieve lower latency and higher throughput for faster packet processing in the user space. DPDK uses the SR-IOV network for hardware-based I/O sharing.
+
+.Prerequisites
+* Your cluster is configured to run DPDK workloads.
+* You have created a project on the cluster in which the VM is running.
+
+.Procedure
+. Edit the `VirtualMachine` manifest to include information about the SR-IOV network interface, CPU topology, CRI-O annotations, and huge pages:
++
+.Example `VirtualMachine` manifest
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: rhel-dpdk-vm
+spec:
+  running: true
+  template:
+    metadata:
+      annotations:
+        cpu-load-balancing.crio.io: disable <1>
+        cpu-quota.crio.io: disable <2>
+        irq-load-balancing.crio.io: disable <3>
+    spec:
+      domain:
+        cpu:
+          sockets: 1 <4>
+          cores: 5 <5>
+          threads: 2
+          dedicatedCpuPlacement: true
+        interfaces:
+          - masquerade: {}
+            name: default
+          - model: virtio
+            name: nic-east
+            pciAddress: '0000:07:00.0'
+            sriov: {}
+          networkInterfaceMultiqueue: true
+          rng: {}
+      memory:
+        hugepages:
+          pageSize: 1Gi <6>
+      resources:
+        requests:
+          memory: 8Gi
+      networks:
+        - name: default
+          pod: {}
+        - multus:
+            networkName: dpdk-net <7>
+          name: nic-east
+#...
+----
+<1> This annotation specifies that load balancing is disabled for CPUs that are used by the container.
+<2> This annotation specifies that the CPU quota is disabled for CPUs that are used by the container.
+<3> This annotation specifies that Interrupt ReQuest (IRQ) load balancing is disabled for CPUs that are used by the container.
+<4> The number of sockets inside the VM. This field must be set to `1` for the CPUs to be scheduled from the same Non-Uniform Memory Access (NUMA) node.
+<5> The number of cores inside the VM. This must be a value greater than or equal to `1`. In this example, the VM is scheduled with 5 hyper-threads or 10 CPUs.
+<6> The size of the huge pages. The possible values for x86-64 architecture are 1Gi and 2Mi. In this example, the request is for 8 huge pages of size 1Gi. 
+<7> The name of the SRI-OV `NetworkAttachmentDefinition` object.
+
+. Save and exit the editor.
+. Configure the guest operating system. The following example shows the configuration steps for RHEL8 OS:
+.. Configure isolated CPUs and specify huge pages by using the GRUB bootloader command line interface:
++
+[source,terminal]
+----
+$ grubby --update-kernel=ALL --args="default_hugepagesz=1GB hugepagesz=1G hugepages=8 isolcpus=2-9"
```

Comment:
The values use here and in the following commands need to match some of the attributes earlier used in the VM spec, used in other commands of this section (and maybe even in the PerformanceProfile). We should call those out. @orelmisan would you please help defining the mapping?

Reply:
@orelmisan Do you have any suggested changes for this section?

Reply:
Commented in the thread above. We can close this one.